ome/bioformats
# Matlab utility functions

Improve utility functions so that high-level Matlab functions can be used as painlessly as possible. 
In particular create a function to check/add loci_tools.jar in the Java class path.
### Testing scenarios

For all tests, make sure loci_tools.jar is in the Matlab path, restart Matlab (clear all paths) and perform the test twice.
NB: to use `runtests`, you need the xUnit test framework for Matlab to be installed and added to the path (http://www.mathworks.com/matlabcentral/fileexchange/22846-matlab-xunit-test-framework)
1. Add loci_tools.jar to Matlab static Java class path (using `edit classpath.txt`) and restart
   1. Run `bfopen`: should work
   2. Run `r=bfGetReader(/path/to/file)`: return r
   3. Run TestCheckJavaPath with `runtests`: all tests should pass
2. Remove loci_tools.jar from the Matlab static Java class path and restart
   1. Set `autoloadBioformats=0` in bfopen.m and run `bfopen`: should fail with an informative error messaage
   2. Set `autoloadBioformats=1` in bfopen.m and run `bfopen`: should work
   3. Run `r=bfGetReader(/path/to/file)`: return r
   4. Run TestCheckJavaPath with `runtests`: all tests should pass
3. Run `tic; bfCheckJavaPath(); toc`. Check the completion time is less than .2 s.
